src/community/script/core/src/main/java/org/geoserver/script/rest/service/AppService.java

```diff
@@ -0,0 +1,107 @@
+ **/
+package org.geoserver.script.rest.service;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.rest.RestException;
+import org.geoserver.script.ScriptManager;
+import org.geoserver.script.app.AppHook;
+import org.geotools.util.logging.Logging;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import javax.script.ScriptEngine;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.lang.String.format;
+
+@Service
+public class AppService {
+    static Logger LOGGER = Logging.getLogger(AppService.class);
+
+    @Autowired
+    ScriptManager scriptManager;
+
+    public ResponseEntity<?> getAppList() {
+        org.geoserver.platform.resource.Resource appRoot = scriptManager.app();
+        List<Resource> apps = Resources.list(appRoot, Resources.DirectoryFilter.INSTANCE);
+
+        JSONArray array = new JSONArray();
+        for (org.geoserver.platform.resource.Resource f : apps) {
+            array.add(f.name());
+        }
+
+        JSONObject obj = new JSONObject();
+        obj.put("scripts", array);
+
+        return new ResponseEntity<String>(obj.toString(2), HttpStatus.OK);
+    }
+
+    public void getApp(HttpServletRequest request, HttpServletResponse response, String app) {
+        org.geoserver.platform.resource.Resource script = null;
+        try {
+            org.geoserver.platform.resource.Resource appDir;
+            try {
+                appDir = scriptManager.app(app);
+            } catch (IllegalStateException e) {
+                throw new RestException(format("Error looking up app directory %s", app),
+                        HttpStatus.INTERNAL_SERVER_ERROR);
+            }
+
+            if (appDir == null) {
+                throw new RestException(format("No such app %s", app), HttpStatus.NOT_FOUND);
+            }
+
+            //look for main script
+            script = scriptManager.findAppMainScript(appDir);
+            if (script == null) {
+                throw new RestException(format("No main file for app %s", app), HttpStatus.NOT_FOUND);
+            }
+
+            ScriptEngine eng = scriptManager.createNewEngine(script);
+            if (eng == null) {
+                throw new RestException(format("Script engine for %s not found", script.name()), HttpStatus.BAD_REQUEST);
+            }
+
+            //look up the app hook
+            AppHook hook = scriptManager.lookupAppHook(script);
+            if (hook == null) {
+                //TODO: fall back on default
```

Comment:
Has this TODO been resolved?

Reply:
The TODO is an artifact from the original version, not sure what default was in this context.

Reply:
This is an artifact from the original code.  Not sure what was meant here.  Can remove TODO if you think that is better.

Reply:
If it is left over from the original code, I guess it is safer to keep it. Probably won't make much difference either way.